lib/apiservers/service/restapi/handlers/vch_get.go

```diff
@@ -0,0 +1,412 @@
+		principal.(Credentials).pass,
+		params.Thumbprint,
+		nil,
+		nil)
+
+	d.ID = params.VchID
+
+	vch, err := getVCH(params.HTTPRequest.Context(), d)
+
+	if err != nil {
+		return operations.NewGetTargetTargetVchVchIDDefault(util.StatusCode(err)).WithPayload(&models.Error{Message: err.Error()})
+	}
+
+	return operations.NewGetTargetTargetVchVchIDOK().WithPayload(vch)
+}
+
+func (h *VCHDatacenterGet) Handle(params operations.GetTargetTargetDatacenterDatacenterVchVchIDParams, principal interface{}) middleware.Responder {
+	d := buildData(
+		url.URL{Host: params.Target},
+		principal.(Credentials).user,
+		principal.(Credentials).pass,
+		params.Thumbprint,
+		&params.Datacenter,
+		nil)
+
+	d.ID = params.VchID
+
+	vch, err := getVCH(params.HTTPRequest.Context(), d)
+
+	if err != nil {
+		return operations.NewGetTargetTargetDatacenterDatacenterVchVchIDDefault(util.StatusCode(err)).WithPayload(&models.Error{Message: err.Error()})
+	}
+
+	return operations.NewGetTargetTargetDatacenterDatacenterVchVchIDOK().WithPayload(vch)
+}
+
+func getVCH(ctx context.Context, d *data.Data) (*models.VCH, error) {
+	validator, err := validateTarget(ctx, d)
+	if err != nil {
+		return nil, util.WrapError(400, err)
+	}
+
+	executor := management.NewDispatcher(validator.Context, validator.Session, nil, false)
+	vch, err := executor.NewVCHFromID(d.ID)
+	if err != nil {
+		return nil, util.NewError(500, fmt.Sprintf("Failed to inspect VCH: %s", err))
+	}
+
+	err = validate.SetDataFromVM(validator.Context, validator.Session.Finder, vch, d)
+	if err != nil {
+		return nil, util.NewError(500, fmt.Sprintf("Failed to load VCH data: %s", err))
+	}
+
+	model, err := vchToModel(vch, d, executor)
+	if err != nil {
+		return nil, util.WrapError(500, err)
+	}
+
+	return model, nil
+}
+
+func vchToModel(vch *vm.VirtualMachine, d *data.Data, executor *management.Dispatcher) (*models.VCH, error) {
+	vchConfig, err := executor.GetNoSecretVCHConfig(vch)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to retrieve VCH information: %s", err)
+	}
+
+	model := &models.VCH{}
+	model.Version = models.Version(vchConfig.Version.ShortVersion())
+	model.Name = vchConfig.Name
+
+	// compute
+	model.Compute = &models.VCHCompute{
+		CPU: &models.VCHComputeCPU{
+			Limit:       asMHz(d.ResourceLimits.VCHCPULimitsMHz),
+			Reservation: asMHz(d.ResourceLimits.VCHCPUReservationsMHz),
+			Shares:	     asShares(d.ResourceLimits.VCHCPUShares),
+		},
+		Memory: &models.VCHComputeMemory{
+			Limit:       asMiB(d.ResourceLimits.VCHMemoryLimitsMB),
+			Reservation: asMiB(d.ResourceLimits.VCHMemoryReservationsMB),
+			Shares:	     asShares(d.ResourceLimits.VCHMemoryShares),
+		},
+		Resource: &models.ManagedObject{
+			ID: vchConfig.Container.ComputeResources[0].String(), // TODO: This doesn't seem to be an ID?
+		},
+	}
+
+	// network
+	model.Network = &models.VCHNetwork{
+		Bridge: &models.VCHNetworkBridge{
+			PortGroup: &models.ManagedObject{
+				ID: vchConfig.Network.BridgeNetwork,
```

Comment:
did you test it? feels it's not right
you might reference here: https://github.com/vmware/vic/blob/master/lib/install/validate/config_to_data.go#L366
You could also reference the code in the above file for how to read back VCH configuration

Reply:
Good catch! It looks like it's always returning "bridge", which happened to be the name of my bridge network, so I didn't notice.